### ISSUE TITLE
[Bug] macOS 15 Local Network Privacy blocks Chia client

### ISSUE THREAD

**Issue by jamesic** [Chia-Network/chia-blockchain#18650]
### What happened?

After upgrading the macOS to the latest version 15, I got the below error from the log:

```
2024-09-28T23:39:41.536 harvester harvester_server        : INFO     Cannot connect to host 192.168.1.92:8447 ssl:<ssl.SSLContext object at 0x10acfd5c0> [No route to host]
2024-09-28T23:39:41.536 harvester harvester               : INFO     Failed to connect to PeerInfo(_ip=IPv4Address('192.168.1.92'), _port=8447)
```

It seems the issue is related to the article https://youtrack.jetbrains.com/articles/SUPPORT-A-564/macOS-15-Local-Network-Privacy-blocks-IDE-connections-to-local-resources-databases-app-servers-etc-unless-explicitly-permitted

Is it possible to update the Chia client to request the permission of "Local Network Privacy" so that I can continue to use the app? Thank you.

### Version

2.4.3

### What platform are you using?

macOS

### What ui mode are you using?

CLI

### Relevant log output or stacktrace

```shell
2024-09-28T23:39:41.531 harvester harvester               : INFO     Starting service harvester ...
2024-09-28T23:39:41.532 harvester harvester               : INFO     chia-blockchain version: 2.4.3
2024-09-28T23:39:41.532 harvester harvester_server        : INFO     Service capabilities: [(1, '1'), (2, '1'), (3, '1')]
2024-09-28T23:39:41.534 harvester harvester               : INFO     Started harvester service on network_id: mainnet at port None
2024-09-28T23:39:41.535 harvester harvester               : INFO     Add resolved PeerInfo(_ip=IPv4Address('192.168.1.92'), _port=8447)
2024-09-28T23:39:41.536 harvester harvester_server        : INFO     Cannot connect to host 192.168.1.92:8447 ssl:<ssl.SSLContext object at 0x10acfd5c0> [No route to host]
2024-09-28T23:39:41.536 harvester harvester               : INFO     Failed to connect to PeerInfo(_ip=IPv4Address('192.168.1.92'), _port=8447)
```

**Comment by ChiaMineJP** [comment]
I upgraded my MBP to macOS 15 and found that macOS automaticall adds an application which tries to access to a local network ip address to the list of `Privacy & Security -> Local Network`.

For instance, when I ran `ping 192.168.xxx.xxx` on my `iTerm` terminal app, allow-deny selection dialog appears and at the same time `iTerm` app was added to the list of `Privacy & Security -> Local Network`.

So I believe there is no special program code to add an application to the list. The responsibility to find such an application should be macOS 15 itself.

**Comment by ChiaMineJP** [comment]
![Screenshot 2024-09-29 at 21 55 50](https://github.com/user-attachments/assets/5ac21734-ed2e-421e-8243-ee56a83fa083)

I tried with source install and the dialog appeared.
After choosing `Allow`, PyCharm was added to the list in the `Local Network` automatically.

<img width="495" alt="image" src="https://github.com/user-attachments/assets/fc0d8565-de32-4429-9c52-07cc488fcacd">

**Comment by jamesic** [comment]
How to add the Chia app to the Local Network Privacy list?

I tried opening the Chia GUI app and executing the Chia command, but none of the approaches allowed me to add the Chia app to the Local Network Privacy list.

**Comment by ChiaMineJP** [comment]
When I opened the GUI (package install) in macOS 15, the popup dialog below appeared

![Screenshot 2024-09-30 at 1 05 23](https://github.com/user-attachments/assets/acdca81d-cc22-4970-8aa6-eac58603f504)

I also found that `Electron` is shown in the list of `Privacy & Security -> Local Network`.

![image](https://github.com/user-attachments/assets/b14dde29-8d8b-431b-a2b6-b1430fc136c6)

Did you see `Electron` in `Privacy & Security -> Local Network`?

**Comment by jamesic** [comment]
The issue was resolved after I renamed the config.yaml and manually merged the config with the latest version generated automatically. Thank you for your support.

**Comment by jamesic** [comment]
The issue was not completely resolved. It appeared to be resolved for a few minutes and the error appeared again once the chia harvester was running in the background. I was able to get the Chia App added to the Local Network, but it didn't help for the below command which might be running in the background.
```bash
chia start harvester -r
```
<img width="483" alt="Screenshot 2024-10-02 at 8 15 47 pm" src="https://github.com/user-attachments/assets/c60ee1a4-cae7-48bf-b986-61682034de68">

**Comment by BrandtH22** [comment]
Hey @jamesic , for the harvester section of the config file are you defining the port as the lan port as shown in the nc command or via the localhost ip/hostname? (i.e. `localhost` or `127.0.0.1`)

**Comment by jamesic** [comment]
> Hey @jamesic , for the harvester section of the config file are you defining the port as the lan port as shown in the nc command or via the localhost ip/hostname? (i.e. `localhost` or `127.0.0.1`)

Below is my harvester config:
```yaml
harvester:
  chia_ssl_ca:
    crt: config/ssl/ca/chia_ca.crt
    key: config/ssl/ca/chia_ca.key
  decompressor_thread_count: 0
  decompressor_timeout: 20
  disable_cpu_affinity: false
  enforce_gpu_index: false
  farmer_peers:
  - host: 192.168.1.92
    port: 8447
```

**Comment by BrandtH22** [comment]
Hey @jamesic , as a heads up we can generally provide more timely and thorough support in our discord server (https://discord.gg/chia).

To ensure I understand the setup correctly, can you confirm the below:
- you have two different machines, both have been recently updated to 2.4.3
- new config files have been created for both machines
- the CA files from the farming machine have been copied and added to the harvester machine with the command `chia init -c <directory>`
- the farming and full node machine has the ip of `192.168.1.92`
- the farming machine and harvester have been restarted


My confusion arises from the comment of `It appeared to be resolved for a few minutes and the error appeared again once the chia harvester was running in the background.`
What are the two different ways you are running the harvester that have different results?

**Comment by jamesic** [comment]
@BrandtH22 , yes, that's the correct setup. There are no two different ways. I just started the harvester using the below command from Terminal.
```bash
chia start harvester -r
```
As a temp workaround, I started a Debian 12 VM from VMware Fusion to harvest the plots. However, it can only harvest for 30 minutes and then the entire VM is disconnected. I created a separate post at https://community.broadcom.com/vmware-cloud-foundation/discussion/vm-is-disconnected-automatically-after-30-minutes-in-macos-15 for the VM issue, but I feel it might also related to the Local Network Privacy in macOS 15.
